rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/RestSecurity.java

```diff
@@ -0,0 +1,97 @@
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+package org.codice.ddf.cxf;
+
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import org.apache.cxf.common.util.Base64Utility;
+import org.apache.cxf.jaxrs.client.Client;
+import org.apache.cxf.rs.security.saml.DeflateEncoderDecoder;
+import org.apache.ws.security.WSSecurityException;
+import org.apache.ws.security.saml.ext.AssertionWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Cookie;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Provides methods that help with securing RESTful (jaxrs) communications.
+ */
+public final class RestSecurity {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RestSecurity.class);
+
+    /**
+     * Parses the incoming subject for a saml assertion and sets that as a cookie on the client.
+     *
+     * @param subject Subject containing a SAML-based security token.
+     * @param client  Non-null client to set the cookie on.
+     * @throws NullPointerException if client is null
+     */
+    public static void setSubjectOnClient(Subject subject, Client client) {
+        if (subject != null) {
+            javax.ws.rs.core.Cookie cookie = createSamlCookie(subject);
+            if (cookie == null) {
+                LOGGER.debug("SAML Cookie was null. Unable to set the cookie for the client.");
+                return;
+            }
+            client.cookie(cookie);
+        }
+    }
+
+    /**
+     * Creates a cookie to be returned to the browser if the token was successfully exchanged for
+     * a SAML assertion.
+     *
+     * @param subject - {@link ddf.security.Subject} to create the cookie from
+     */
+    private static Cookie createSamlCookie(Subject subject) {
+        Cookie cookie = null;
+        org.w3c.dom.Element samlToken = null;
+        try {
+            for (Object principal : subject.getPrincipals().asList()) {
+                if (principal instanceof SecurityAssertion) {
+                    samlToken = ((SecurityAssertion) principal).getSecurityToken()
+                            .getToken();
+                }
+            }
+            if (samlToken != null) {
+                cookie = new Cookie(SecurityConstants.SAML_COOKIE_NAME, encodeSaml(samlToken));
```

Comment:
@stustison so debugging is definitely one scenario (and this obviously aligns with the reasons to have the ability Disable the CN Check which should only be used for debugging).

However if you are inside your trusted zone and not going outside of your trusted perimeter, many people prefer to only use HTTP when communicating between 2 software instances in there.  This is a pretty standard option and where/why many folks use HTTPS Termination points at the external boundry.  The Assertion is still signed, so it can't be changed.

On a slight tangent I would love to see DDF not force ANY security requirements on people who use it.  But that may be my misunderstanding and me viewing DDF as a flexible/customizable "Framework" vs. more of a Software Product that has certain pluggable points.